Build failure: dotnetCorePackages.sdk_10_0

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Run `nix shell --print-build-logs nixpkgs#dotnetCorePackages.dotnet_10.vmr`

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

Cannot confirm if hydra build is present/working. Search gives 500 error on usage

### Relevant log output

```console
dotnet-stage0-vmr>   [17:32:39.93] Building aspnetcore
dotnet-stage0-vmr>   Running command:
dotnet-stage0-vmr>     /build/dotnet-10.0.0-preview.6.25358.103/src/aspnetcore/eng/build.sh --restore --build --pack --publish --no-build-java --build-installers --build-managed  --sourceBuild --fromVMR --configuration Release --verbosity minimal /p:TargetRid=linux-x64 /p:OfficialBuildId=20250708.103 /p:DotNetPackageVersionPropsPath=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/obj/PackageVersions/PackageVersions.aspnetcore.props /p:SourceBuiltAssetsDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/assets/Release/ /p:SourceBuiltAssetManifestsDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/obj/manifests/Release/aspnetcore/ /p:RepositoryName=aspnetcore /p:RepositoryUrl=https://github.com/dotnet/dotnet /p:PreviouslySourceBuiltNupkgCacheDir="/nix/store/plmpm277h4999nzmar40ylaicn5l1wwa-Private.SourceBuilt.Artifacts.10.0.100-preview.6.25302.104.centos.9-x64/" /p:ReferencePackageNupkgCacheDir="/build/dotnet-10.0.0-preview.6.25358.103/prereqs/packages/reference/" /p:TrackPrebuiltUsageReportDir="/build/dotnet-10.0.0-preview.6.25358.103/artifacts/log/Release/aspnetcore/" /p:TargetOS=linux /p:SourceBuiltShippingPackagesDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/packages/Release/Shipping/aspnetcore/ /p:SourceBuiltNonShippingPackagesDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/packages/Release/NonShipping/aspnetcore/ /p:SourceBuiltPdbArtifactsDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/SymStore/Release/aspnetcore/ /p:SourceBuiltArtifactsBasePath=/build/dotnet-10.0.0-preview.6.25358.103/ /p:UsingToolMicrosoftNetCompilers=false /p:PortableBuild=true /p:BaseRid=linux-x64 /p:RepoOriginalSourceRevisionId=a1b8651524b3c7005f5a5bbc8d0646a3266eb608  -bl --arch x64 /p:TargetRuntimeIdentifier=linux-x64 /p:PublicBaseURL=file:///build/dotnet-10.0.0-preview.6.25358.103/artifacts/assets/Release/ /p:CrossArchitectureInstallerBasePath=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/assets/Release/ /p:EnablePackageValidation=false
dotnet-stage0-vmr>     Log: /build/dotnet-10.0.0-preview.6.25358.103/artifacts/log/Release/aspnetcore/aspnetcore.log
dotnet-stage0-vmr>     With Environment Variables:
dotnet-stage0-vmr>       DOTNET_INSTALL_DIR=/build/dotnet-10.0.0-preview.6.25358.103/.dotnet/
dotnet-stage0-vmr>       DOTNET_PATH=/build/dotnet-10.0.0-preview.6.25358.103/.dotnet/
dotnet-stage0-vmr>       DOTNET_HOST_PATH=/build/dotnet-10.0.0-preview.6.25358.103/.dotnet/dotnet
dotnet-stage0-vmr>       _InitializeDotNetCli=/build/dotnet-10.0.0-preview.6.25358.103/.dotnet
dotnet-stage0-vmr>       _DotNetInstallDir=/build/dotnet-10.0.0-preview.6.25358.103/.dotnet/
dotnet-stage0-vmr>       _InitializeToolset=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/source-built-sdks/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj
dotnet-stage0-vmr>       DeterministicSourcePaths=true
dotnet-stage0-vmr>       SourceRoot=/build/dotnet-10.0.0-preview.6.25358.103/src/aspnetcore/
dotnet-stage0-vmr>       RestoreConfigFile=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/obj/aspnetcore/NuGet.config
dotnet-stage0-vmr>       NUGET_PACKAGES=/build/dotnet-10.0.0-preview.6.25358.103/src/aspnetcore/artifacts/.packages/
dotnet-stage0-vmr>       NuGetAudit=false
dotnet-stage0-vmr>       MSBUILDADDITIONALSDKRESOLVERSFOLDER=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/toolset/VSSdkResolvers/
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_ID_ARCADE=Microsoft.DotNet.Arcade.Sdk
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_ID_ARCADE_SHARED_FX_SDK=Microsoft.DotNet.SharedFramework.Sdk
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_ID_ARCADE_CMAKE_SDK=Microsoft.DotNet.CMake.Sdk
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_ID_TRAVERSAL=Microsoft.Build.Traversal
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_ID_NOTARGETS=Microsoft.Build.NoTargets
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_VERSION_ARCADE=10.0.0-beta.25358.103
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_VERSION_ARCADE_SHARED_FX_SDK=10.0.0-beta.25358.103
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_VERSION_ARCADE_CMAKE_SDK=10.0.0-beta.25358.103
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_VERSION_TRAVERSAL=3.4.0
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_VERSION_NOTARGETS=3.7.0
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_DIR_ARCADE=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/source-built-sdks/Microsoft.DotNet.Arcade.Sdk/
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_DIR_ARCADE_SHARED_FX_SDK=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/source-built-sdks/Microsoft.DotNet.SharedFramework.Sdk/
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_DIR_ARCADE_CMAKE_SDK=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/source-built-sdks/Microsoft.DotNet.CMake.Sdk/
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_DIR_TRAVERSAL=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/source-built-sdks/Microsoft.Build.Traversal/
dotnet-stage0-vmr>       SOURCE_BUILT_SDK_DIR_NOTARGETS=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/source-built-sdks/Microsoft.Build.NoTargets/
dotnet-stage0-vmr>   DirSize Before Building aspnetcore
dotnet-stage0-vmr>   Filesystem      Size  Used Avail Use% Mounted on
dotnet-stage0-vmr>   /dev/sdf       1007G   29G  928G   3% /build
dotnet-stage0-vmr> /build/dotnet-10.0.0-preview.6.25358.103/repo-projects/Directory.Build.targets(544,5): error MSB3073: The command "/build/dotnet-10.0.0-preview.6.25358.103/src/aspnetcore/eng/build.sh --restore --build --pack --publish --no-build-java --build-installers --build-managed  --sourceBuild --fromVMR --configuration Release --verbosity minimal /p:TargetRid=linux-x64 /p:OfficialBuildId=20250708.103 /p:DotNetPackageVersionPropsPath=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/obj/PackageVersions/PackageVersions.aspnetcore.props /p:SourceBuiltAssetsDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/assets/Release/ /p:SourceBuiltAssetManifestsDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/obj/manifests/Release/aspnetcore/ /p:RepositoryName=aspnetcore /p:RepositoryUrl=https://github.com/dotnet/dotnet /p:PreviouslySourceBuiltNupkgCacheDir="/nix/store/plmpm277h4999nzmar40ylaicn5l1wwa-Private.SourceBuilt.Artifacts.10.0.100-preview.6.25302.104.centos.9-x64/" /p:ReferencePackageNupkgCacheDir="/build/dotnet-10.0.0-preview.6.25358.103/prereqs/packages/reference/" /p:TrackPrebuiltUsageReportDir="/build/dotnet-10.0.0-preview.6.25358.103/artifacts/log/Release/aspnetcore/" /p:TargetOS=linux /p:SourceBuiltShippingPackagesDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/packages/Release/Shipping/aspnetcore/ /p:SourceBuiltNonShippingPackagesDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/packages/Release/NonShipping/aspnetcore/ /p:SourceBuiltPdbArtifactsDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/SymStore/Release/aspnetcore/ /p:SourceBuiltArtifactsBasePath=/build/dotnet-10.0.0-preview.6.25358.103/ /p:UsingToolMicrosoftNetCompilers=false /p:PortableBuild=true /p:BaseRid=linux-x64 /p:RepoOriginalSourceRevisionId=a1b8651524b3c7005f5a5bbc8d0646a3266eb608  -bl --arch x64 /p:TargetRuntimeIdentifier=linux-x64 /p:PublicBaseURL=file:///build/dotnet-10.0.0-preview.6.25358.103/artifacts/assets/Release/ /p:CrossArchitectureInstallerBasePath=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/assets/Release/ /p:EnablePackageValidation=false > /build/dotnet-10.0.0-preview.6.25358.103/artifacts/log/Release/aspnetcore/aspnetcore.log 2>&1" exited with code 1. [/build/dotnet-10.0.0-preview.6.25358.103/repo-projects/aspnetcore.proj]
dotnet-stage0-vmr>   warning: Building of C# project is enabled and has dependencies on NodeJS projects. Building of NodeJS projects is enabled since node is detected on PATH.
dotnet-stage0-vmr>   warning: Note that if you are running Source Build, building NodeJS projects will be disabled later on.
dotnet-stage0-vmr>     Determining projects to restore...
dotnet-stage0-vmr>     Restored /build/dotnet-10.0.0-preview.6.25358.103/artifacts/source-built-sdks/Microsoft.DotNet.Arcade.Sdk/tools/Tools.proj (in 115 ms).
dotnet-stage0-vmr>     Determining projects to restore...
dotnet-stage0-vmr>     Restored /build/dotnet-10.0.0-preview.6.25358.103/src/aspnetcore/eng/tools/GenerateFiles/GenerateFiles.csproj (in 11 ms).
dotnet-stage0-vmr>     /nix/store/1cxc32pmrlqwnmd9v6alnwjsy6yhzr56-patch-nupkgs/bin/patch-nupkgs: line 13: cd: null directory
dotnet-stage0-vmr>   /nix/store/0blarqb00ks2wayr3w6v8vj9yy52dn05-patch-restored-packages.proj(6,5): error MSB3073: The command "patch-nupkgs '''' 2>&1" exited with code 1. [/build/dotnet-10.0.0-preview.6.25358.103/src/aspnetcore/eng/tools/GenerateFiles/GenerateFiles.csproj]
dotnet-stage0-vmr>   Build failed with exit code 1. Check errors above.
dotnet-stage0-vmr>   'aspnetcore' failed during build.
dotnet-stage0-vmr>   See '/build/dotnet-10.0.0-preview.6.25358.103/artifacts/log/Release/aspnetcore/aspnetcore.log' for more information.
dotnet-stage0-vmr> Build FAILED.
dotnet-stage0-vmr> /build/dotnet-10.0.0-preview.6.25358.103/repo-projects/Directory.Build.targets(544,5): error MSB3073: The command "/build/dotnet-10.0.0-preview.6.25358.103/src/aspnetcore/eng/build.sh --restore --build --pack --publish --no-build-java --build-installers --build-managed  --sourceBuild --fromVMR --configuration Release --verbosity minimal /p:TargetRid=linux-x64 /p:OfficialBuildId=20250708.103 /p:DotNetPackageVersionPropsPath=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/obj/PackageVersions/PackageVersions.aspnetcore.props /p:SourceBuiltAssetsDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/assets/Release/ /p:SourceBuiltAssetManifestsDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/obj/manifests/Release/aspnetcore/ /p:RepositoryName=aspnetcore /p:RepositoryUrl=https://github.com/dotnet/dotnet /p:PreviouslySourceBuiltNupkgCacheDir="/nix/store/plmpm277h4999nzmar40ylaicn5l1wwa-Private.SourceBuilt.Artifacts.10.0.100-preview.6.25302.104.centos.9-x64/" /p:ReferencePackageNupkgCacheDir="/build/dotnet-10.0.0-preview.6.25358.103/prereqs/packages/reference/" /p:TrackPrebuiltUsageReportDir="/build/dotnet-10.0.0-preview.6.25358.103/artifacts/log/Release/aspnetcore/" /p:TargetOS=linux /p:SourceBuiltShippingPackagesDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/packages/Release/Shipping/aspnetcore/ /p:SourceBuiltNonShippingPackagesDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/packages/Release/NonShipping/aspnetcore/ /p:SourceBuiltPdbArtifactsDir=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/SymStore/Release/aspnetcore/ /p:SourceBuiltArtifactsBasePath=/build/dotnet-10.0.0-preview.6.25358.103/ /p:UsingToolMicrosoftNetCompilers=false /p:PortableBuild=true /p:BaseRid=linux-x64 /p:RepoOriginalSourceRevisionId=a1b8651524b3c7005f5a5bbc8d0646a3266eb608  -bl --arch x64 /p:TargetRuntimeIdentifier=linux-x64 /p:PublicBaseURL=file:///build/dotnet-10.0.0-preview.6.25358.103/artifacts/assets/Release/ /p:CrossArchitectureInstallerBasePath=/build/dotnet-10.0.0-preview.6.25358.103/artifacts/assets/Release/ /p:EnablePackageValidation=false > /build/dotnet-10.0.0-preview.6.25358.103/artifacts/log/Release/aspnetcore/aspnetcore.log 2>&1" exited with code 1. [/build/dotnet-10.0.0-preview.6.25358.103/repo-projects/aspnetcore.proj]
dotnet-stage0-vmr>     0 Warning(s)
dotnet-stage0-vmr>     1 Error(s)
dotnet-stage0-vmr> Time Elapsed 00:16:52.85
dotnet-stage0-vmr> Build failed with exit code 1. Check errors above.
error: builder for '/nix/store/8avb0rm8samjdxaykj9nz2qqxvd6gfby-dotnet-stage0-vmr-10.0.0-preview.6.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/ba8sg9gq21aqywyha3n43iz69ksgp9ax-dotnet-stage0-sdk-10.0.100-preview.6.25358.103.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ls3spbw0k616w42c1pw26ib3cd1pwp3d-dotnet-vmr-10.0.0-preview.6.drv' failed to build
```

### Additional context

Build logs are extremely length, only posted context of the error.

### System metadata

Running off Ubuntu

dominion@CYBERSTATIONXVI:~$ sudo nix shell nixpkgs#nix-info
root@CYBERSTATIONXVI:/home/dominion# nix-info
system: "x86_64-linux", multi-user?: no, version: nix-env (Nix) 2.18.1, nixpkgs: not found

### Notify maintainers

@corngood 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Just working on developing some software for the .NET 10 release at the end of the year. Trying to get the nix builds working but `dotnetCorePackages.sdk_10_0` appears to be dysfunctional.

Not a big rush, I'm guessing it'll be fixed by the time of the release. Would be nice to have the previews working if possible though.

--- a/pkgs/development/compilers/dotnet/vmr.nix
+++ b/pkgs/development/compilers/dotnet/vmr.nix
@@ -230,6 +230,8 @@ stdenv.mkDerivation rec {
       substituteInPlace \
         src/runtime/src/coreclr/ilasm/CMakeLists.txt \
         --replace-fail 'set_source_files_properties( prebuilt/asmparse.cpp PROPERTIES COMPILE_FLAGS "-O0" )' ""
+    ''
+    + lib.optionalString (lib.versionOlder version "10") ''
 
       # https://github.com/dotnet/source-build/issues/4444
       xmlstarlet ed \
@@ -237,8 +239,6 @@ stdenv.mkDerivation rec {
         -s '//Project/Target/MSBuild[@Targets="Restore"]' \
         -t attr -n Properties -v "NUGET_PACKAGES='\$(CurrentRepoSourceBuildPackageCache)'" \
         src/aspnetcore/eng/Tools.props
-    ''
-    + lib.optionalString (lib.versionOlder version "10") ''
       # patch packages installed from npm cache
       xmlstarlet ed \
         --inplace \

